### PR TITLE
initial work on requirements chapter

### PR DESCRIPTION
**Pull request by AcidSepp**

### CODE REVIEW

File: LaTeX/Parts/Chapter7.tex [hunk 1]
```diff
@@ -0,0 +1,169 @@
+\chapter{Requirements}
+The following sections list the requirements and their respective priorities.
+In addition, it is indicated whether and how they are currently implemented.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
Additionally it is indicated whether and how they are currently implemented.
```

File: LaTeX/Parts/Chapter7.tex [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+\section{Non-Functional Requirements}
+
+\subsection{Transparency (Must Have High)}\label{sec:Transparency}
+\emph{Varroa has to be comprehensible for the user.}
+
+This requirement is fulfilled.
+Varroa implements expressive logging with configurable levels and outputs.
+This enables rich insight in the execution process of a Varroa test.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
This enables a rich insight in the execution process of a Varroa test.
```
Don't know whether this is an improvement, but the sentence feels incomplete.

File: LaTeX/Parts/Chapter7.tex [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+\subsection{10.000.000 MQTT Clients (Must Have High)} 
+\emph{Varroa has to be able to generate a large amount of clients.}
+
+The fulfilment of this requirement is not verified yet.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex:17]: This sentence feels very incomplete.

File: LaTeX/Parts/Chapter7.tex [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+\subsection{Scalability (Must Have)} 
+\emph{Varroa should scale vertically with relatively low scaling costs.}
+
+This requirement is fulfilled.
+To do so Varroa is organised as a distributed System and can easily be extended with new instances running as Varroa Agents.
+
+\subsection{Determinism (Must Have)} 
+\emph{Varroa has to work in deterministic ways, meaning it should produce the same result for a Scenario every time.}
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
\emph{Varroa has to work in deterministic ways, meaning it should produce the same result for the same Scenario every time.}
```

File: LaTeX/Parts/Chapter7.tex [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+This requirement is fulfilled.
+To ensure this determinism several measures are taken in the distribution and execution of a Scenario (see chapter \ref{sec:Architecture}).
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
To guarantee this determinism several measures are taken, ensuring deterministic distribution and execution of a Scenario (see chapter \ref{sec:Architecture}).
```
Not sure about the semicolon.

File: LaTeX/Parts/Chapter7.tex [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+\subsection{Distributed (Must Have Low)} 
+\emph{Varroa is a distributed System.}
+
+This requirement is fulfilled.
+Varroa is organized as a distributed System, it is composed of a Commander and multiple Agents that can run on different machines.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
Varroa is organized as a distributed System.
It is composed of a Commander and multiple Agents.
These components can be run on separate machines. 
```

File: LaTeX/Parts/Chapter7.tex [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+\emph{Varroa has to be easily usable.}
+
+This requirement is fulfilled.
+The user can easily define custom Scenarios as XML files and execute them (see chapter \ref{sec:ScenarioConcept} and \ref{sec:Execution}).
+
+\subsection{Code Quality (Important)} 
+\emph{Varroa's code quality should be very high.}
+
+This requirement is fulfilled.
+To ensure high code quality several measures are taken.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex:49]:
```suggestion
To ensure the high code quality standards, several measures were taken.
```
Sentence still is pretty generic, we should have a further look at this.

File: LaTeX/Parts/Chapter7.tex [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+For example the consequent use of nullability annotiations and a broad test code coverage.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
An example for these measures is the consequent use of nullability as well as broad test code coverage. 
```

File: LaTeX/Parts/Chapter7.tex [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+Also before new code could be added to the master branch it was reviewed by another team member.
+
+\subsection{Stability (Important)} 
+\emph{Varroa has to run in a stable manner.}
+
+This requirement is fulfilled.
+By the use of integration tests and general unit tests stability is ensured.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
Stability is ensured by the use of integration tests and general unit tests.
```

File: LaTeX/Parts/Chapter7.tex [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+\subsection{Resource efficiency (Important)} 
+\emph{Varroa has to use the available computation and memory resources efficiently.}
+%TODO how was this verified.
+
+\subsection{User / Developer Guide (Somewhat Important)} 
+\emph{Varroa needs a User / Developer Guide.}
+
+This requirement is fulfilled. This documentation educates the user on defining a custom scenario and after that executing it on his Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
This requirement is fulfilled. 
The documentation educates the user on the definition of a custom scenario aswell as the exectution of it on her Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).
```
Georg said we gender everything feminine.

**Review by RobinAtherton — CHANGES_REQUESTED** [review]

File: LaTeX/Parts/Chapter7.tex (hunk 3 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter7.tex:17]: It just says that we could not verify this yet. It is okay imho.

File: LaTeX/Parts/Chapter7.tex (hunk 7 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter7.tex:49]: Because of this there are the following sentences.

File: LaTeX/Parts/Chapter7.tex (hunk 10 above)
Comment (AcidSepp) [inline LaTeX/Parts/Chapter7.tex]:
```suggestion
This requirement is fulfilled.
The documentation educates the user on the definition of a custom scenario aswell as the exectution of it on his Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).
```

**Pull request merged**